### ISSUE TITLE
Add a WP-CLI alias

### ISSUE THREAD

**Issue by johnbillion** [humanmade/altis-local-server#221]
Running a WP-CLI command on default Altis local server requires the following:

```
composer server cli -- [command]
```

This works fine but is a non-standard way of interfacing with WP-CLI. If your project is already set up you can instead add an alias:

```
touch wp-cli.yml && wp cli alias add @local --set-ssh=docker:www-data@xxx --config=project
```

where `xxx` is the name of the PHP container, eg. `foo_php_1`.

This allows you to subsequently run:

```
wp @local [command]
```

The advantage this has is that tab completion works and it's a more familiar, shorter, and memorable interface to WP-CLI.

Is it possible for Altis local server to provide a default alias that is aware of the PHP container name?

---

## Acceptance Criteria

- [x] Add `composer server create-alias` command
- [x] Check for and handle an existing file
- [x] Output instructions/help
- [x] Add documentation

---

### Ready for Work Checklist

Is this ticket ready to be worked on? See
[the Play Book Definition of Ready](https://playbook.hmn.md/play/product/definition-of-ready/)

- [x] Is the title clear?
- [x] Is the description clear and detailed enough?
- [x] Are acceptance criteria listed?
- [x] Have any dependencies been identified? (Optional)
- [x] Have any documentation/playbook changes been identified? (Optional)
- [x] Is an estimate or time box assigned?
- [x] Is a priority label assigned?
- [x] Is this ticket added to a milestone?
- [x] Is this ticket added to an epic? (Optional)

**Comment by johnbillion** [comment]
It should also be possible to not use an alias, and put the `ssh:` config into a `wp-cli.yml` file in the root of the project, but there appears to be a bug in WP-CLI when using the docker scheme that I've not looked into yet.

**Comment by johnbillion** [comment]
https://github.com/wp-cli/wp-cli/issues/5447

**Comment by roborourke** [comment]
I did experiment briefly with this as part of #224 regarding pipe detection. This approach requires WP CLI to be installed on the host machine though, not everyone has that. We could only ensure it by adding it as a dependency of local server and aliasing something like `composer exec wp`. The `docker-compose` scheme worked fine for me in testing but I hadn't tried that many different commands yet.

**Comment by johnbillion** [comment]
I don't think this relates to pipe detection etc, rather this is a bonus for anyone who _does_ have WP-CLI installed locally. The `wp-cli.yml` file in the project root will get picked up when the user uses a `wp` command.

My thinking is that as part of the installation step of an Altis project the alias gets added with the name of the container.

**Comment by roborourke** [comment]
Fair. I guess it wouldn't be too hard to document that difference. Projects would potentially need to `.gitignore` the `wp-cli.yml` file though as it would screw up in production.

If you want to keep playing in the meantime use `docker-compose:php` for your SSH setting.

**Comment by johnbillion** [comment]
Does Altis Cloud put a `wp-cli.yml` file into place on the prod server?

**Comment by johnbillion** [comment]
By the way WP-CLI supports a hierarchy for these config files so it could be wp-cli.local.yml instead: https://make.wordpress.org/cli/handbook/references/config/#config-files

**Comment by roborourke** [comment]
No there's no generated `wp-cli.yml` in the cloud - it would still need to be git ignored. This could perhaps be done using a CLI command though eg. `composer server create-wp-cli-yml` that then prompts / recommends git ignoring it.

**Comment by johnbillion** [comment]
In that case if we only introduce an _alias_ into `wp-cli.yml` then we don't need to ensure it gets ignored by git because it won't affect production unless somebody explicitly tries to run that alias while SSHd into a production server, in which case they'll just get a connection error.

Prompting to ignore the file might be a good idea, but there might also be a benefit to committing this file so all developers have access to this alias. It would need setting up once. Maybe this means it would be better off as a Composer command such as `composer server create-alias`.

**Comment by johnbillion** [comment]
I might have a play around with this as a learning exercise for myself. Not done anything with Composer plugins yet.

**Comment by roborourke** [comment]
Awesome, have at it!

Agree on your proposed name for the command. I think the output from the command should indicate how you would then use it eg:

```
Generating wp-cli.yml...
Done!

You can now run WP CLI commands using the following syntax:
wp @local cli info
```

Other nice thing it could do is check for WP CLI being installed on the host, and if not provide installation instructions or link to the WP CLi install docs
   - I personally do this using `composer global require wp-cli/wp-cli-bundle`
   - `brew install wp-cli` also an easier option but should only be mentioned if user is on a mac

**Comment by rmccue** [comment]
Dug into this, and it's definitely possible; given the docker-compose container names are predictable, you can use them directly in `docker-compose:` config:

```
@local
    ssh: docker-compose:my-project-php
```

(These options aren't properly documented; opened https://github.com/wp-cli/handbook/pull/462 for that.)

Due to https://github.com/wp-cli/wp-cli/issues/5447#issuecomment-1703855396 we can't specify a default alias, although we could support eg `wp @local ...` for example. We probably also don't _want_ to specify this by default, as it'd break running wp-cli on the actual servers.

**Comment by mikelittle** [comment]
I misread this ticket (I thought it was about an alias to the `composer server cli` command so that `composer server wp` would also work) and my 1 estimate was too low. I think it will be a 3.
